File: app/src/main/java/com/lazylee/lzywanandroid/data/entity/SearchHistory.java
package com.lazylee.lzywanandroid.data.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;

@Entity(primaryKeys = {"key"})
public class SearchHistory {

    @NonNull
    private String key = "";
    private Long date;

    @Ignore
    public SearchHistory(@NonNull String key, Long date) {
        this.key = key;
        this.date = date;
    }

    public SearchHistory() {
    }

    public @NonNull String getKey() {
        return this.key;
    }

    public void setKey(@NonNull String key) {
        this.key = key;
    }


    @Override
    public String toString() {
        return "SearchHistory{" +
                ", key='" + key + '\'' +
                ", date='" + date + '\'' +
                '}';
    }

    public Long getDate() {
        return this.date;
    }

    public void setDate(Long date) {
        this.date = date;
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/data/entity/SearchHistory.kt
package com.lazylee.lzywanandroid.data.entity

import androidx.room.Entity
import androidx.room.Ignore

@Entity(primaryKeys = arrayOf("key"))
class SearchHistory {

    var key = ""
    var date: Long? = null

    @Ignore
    constructor(key: String, date: Long?) {
        this.key = key
        this.date = date
    }

    constructor() {}


    override fun toString(): String {
        return "SearchHistory{" +
                ", key='" + key + '\''.toString() +
                ", date='" + date + '\''.toString() +
                '}'.toString()
    }
}
